lineapy/plugins/session_writers.py

```diff
@@ -0,0 +1,238 @@
+        for line in self._get_session_input_parameters_lines(
+            session_artifact
+        ).split("\n"):
+            variable_def = line.strip(" ").rstrip(",")
+            if len(variable_def) > 0:
+                variable_name = variable_def.split("=")[0].strip(" ")
+                value = eval(variable_def.split("=")[1].strip(" "))
+                value_type = type(value)
+                if not (
+                    isinstance(value, int)
+                    or isinstance(value, float)
+                    or isinstance(value, str)
+                    or isinstance(value, bool)
+                ):
+                    raise ValueError(
+                        f"LineaPy only supports primitive types as input parameters for now. "
+                        f"{variable_name} in {variable_def} is a {value_type}."
+                    )
+                if ":" in variable_name:
+                    variable_name = variable_def.split(":")[0]
+                    session_input_variables[variable_name] = InputVariable(
+                        variable_name=variable_name,
+                        value=value,
+                        value_type=value_type,
+                    )
+                else:
+                    session_input_variables[variable_name] = InputVariable(
+                        variable_name=variable_name,
+                        value=value,
+                        value_type=value_type,
+                    )
+        return session_input_variables
+
+    def get_session_function_callblock(
+        self, session_artifact: SessionArtifacts
+    ) -> str:
+        """
+        Return the code to make the call to the session function as
+        `session_function_name(input_parameters)`.
+        """
+        session_function_name = self.get_session_function_name(
+            session_artifact
+        )
+        if session_function_name != "":
+            session_input_parameters = ", ".join(
+                self.get_session_input_parameters_spec(session_artifact).keys()
+            )
+            return f"{session_function_name}({session_input_parameters})"
+        else:
+            return ""
+
+    def get_session_artifact_function_definitions(
+        self,
+        session_artifact: SessionArtifacts,
+        include_non_slice_as_comment=False,
+        indentation=4,
+    ) -> List[str]:
+        """
+        Return the definition of each targeted artifacts calculation
+        functions.
+        """
+        return [
+            coll.get_function_definition(
+                graph=session_artifact.session_graph,
+                include_non_slice_as_comment=include_non_slice_as_comment,
+                indentation=indentation,
+            )
+            for coll in session_artifact.usercode_nodecollections
+        ]
+
+    def get_session_function(
+        self,
+        session_artifact,
+        include_non_slice_as_comment=False,
```

Comment:
This function does not seem to use the `include_non_slice_as_comment` for anything. might be a miss.

Reply:
Fixed.